parser/src/powdr.lalrpop

```diff
@@ -555,58 +555,72 @@ ProductOp: BinaryOperator = {
     "%" => BinaryOperator::Mod,
 }
 
-Power: Box<Expression> = {
-    <start:@L> <left:Term> <op:PowOp> <right:Power> <end:@R> => ctx.to_expr_with_source_ref(BinaryOperation { left, op, right }, start, end),
-    Unary,
+Power<StructOption>: Box<Expression> = {
+    <start:@L> <left:Term<StructOption>> <op:PowOp> <right:Power<StructOption>> <end:@R> => ctx.to_expr_with_source_ref(BinaryOperation { left, op, right }, start, end),
+    Unary<StructOption>,
 }
 
 PowOp: BinaryOperator = {
     "**" => BinaryOperator::Pow,
 }
 
-Unary: Box<Expression> = {
-    <start:@L> <op:PrefixUnaryOp> <expr:PostfixUnary> <end:@R> => ctx.to_expr_with_source_ref(UnaryOperation{op, expr}, start, end),
-    PostfixUnary,
+Unary<StructOption>: Box<Expression> = {
+    <start:@L> <op:PrefixUnaryOp> <expr:PostfixUnary<StructOption>> <end:@R> => ctx.to_expr_with_source_ref(UnaryOperation{op, expr}, start, end),
```

Comment:
I think even here we should unconditionally allow structs. I think otherwise the following (maybe weird but still) example would not parse:
```
match -MyStruct{a: 1} {
...
}
```

Reply:
Oh, I see, that's true! But it should be possible to do this in at least some nested expressions like MatchArm, shouldn't it? Or is the problem there that we don't know where the arm body begins?

Reply:
Ok, but at least inside `[]` and `()` it should be possible to allow structs.

Reply:
So I think that ExpressionList can be defined without a parameter...